controllers/hcpvaultsecretsapp_controller.go

```diff
@@ -314,7 +314,7 @@ func (r *HCPVaultSecretsAppReconciler) SetupWithManager(mgr ctrl.Manager, opts c
 			&secretsv1beta1.SecretTransformation{},
 			NewEnqueueRefRequestsHandlerST(r.referenceCache, nil),
 		).
-		Watches(
+		WatchesMetadata(
```

Comment:
We may want to add some comments for the use of `WatchesMetadata` so we don't forget why it's necessary.